frontend/packages/console-shared/src/components/spotlight/Spotlight.tsx

```diff
@@ -7,9 +7,22 @@ type SpotlightProps = {
   interactive?: boolean;
 };
 
+type HasHiddenAncestor = (el: Element) => boolean;
+
 const Spotlight: React.FC<SpotlightProps> = ({ selector, interactive }) => {
+  const hasHiddenAncestor: HasHiddenAncestor = (el) => {
```

Comment:
Why do you need a function for this? This can easily be moved below line the querySelector usage by using `React.useMemo`.

Reply:
Calculate the element along with this logic inside `useMemo` as suggested.

Reply:
Updated. I actually kept it as a function so that in future case with multiple selectors say [ `selector1`, `selector2`, `selector3` ] we could have run it individually on each. But maybe we could refactor it out when necessary.
Moved it inside useMemo.